packages/template-retail-react-app/app/utils/url.js

```diff
@@ -150,22 +151,77 @@ export const getPathWithLocale = (shortCode, opts = {}) => {
     const isDefaultLocaleOfDefaultSite = shortCode === defaultSite.l10n.defaultLocale
     const isDefaultSite = siteRef === defaultSite.alias || siteRef === defaultSite.id
     // rebuild the url with new locale,
-    const newUrl = buildPathWithUrlConfig(
+    const newUrl = urlTemplateLiteral(
         `${pathname}${search}`,
-        {
-            // By default, as for home page, when the values of site and locale belongs to the default site,
-            // they will be not shown in the url just
-            site:
-                isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef
-                    ? ''
-                    : siteRef || defaultSite.alias || defaultSite.id,
-            locale: isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef ? '' : shortCode
-        },
-        opts
+        // By default, as for home page, when the values of site and locale belongs to the default site,
+        // they will be not shown in the url just
+        isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef
+            ? ''
+            : siteRef || defaultSite.alias || defaultSite.id,
+        isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef ? '' : shortCode
     )
     return newUrl
 }
 
+/**
+ * Generates the URL Template literal (Template string) used to build URLs in the App according
+ * the current selected site/locale and the default App URL configuration.
+ *
+ * @param appConfig Application default configuration.
+ * @param siteUrl Current selected Site object.
+ * @param localeUrl Current selected Locale object.
+ * @returns {function(*, *, *, *=): string} function providing href, site and locale generates a site URL.
+ */
+export const getUrlTemplateLiteral = (appConfig, siteUrl, localeUrl) => {
```

Comment:
A [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) isn't actually a thing that code consumers ever use, it's just an implementation detail, a helpful way of writing strings. We could write identical code without using template literals and it would make no difference to the user.

The name `urlTemplateLiteral` makes me think the value is a string, and `getUrlTemplateLiteral` makes me think it's a function that returns a string. I'd suggest something like `createUrlTemplate` returns `fillUrlTemplate` or `createUrlFromTemplate`, something along those lines.